src/tcp/segment.ml

```diff
@@ -135,9 +136,8 @@ module Rx(Time:Mirage_time.S) = struct
 
   let send_challenge_ack q =
     (* TODO:  rfc5961 ACK Throttling *)
-    (* Is this the correct way trigger an ack? *)
-    if Lwt_mvar.is_empty q.rx_data
-      then Lwt_mvar.put q.rx_data (Some [], Some Sequence.zero)
+    if Lwt_mvar.is_empty q.send_ack
+      then Lwt_mvar.put q.send_ack Sequence.zero
       else Lwt.return_unit
```

Comment:
If I read this correctly, the original `Ack.t` stored in `Flow.flow` (aka `Flow.pcb`) is not reachable through `Segment.Rx.t`, which is a shame, maybe `ack` should have been stored deeper.

So it begs the question why is your `send_ack` not a `Ack.t`, so instead of pushing something directly on a mailbox you could do the same `Ack.pushack` dance that the `Rx.thread` do, something like:

```ocaml
  let send_challenge_ack q =
    (* TODO:  rfc5961 ACK Throttling *)
    Ack.pushack q.ack Sequence.zero
```

Reply:
I am not familiar with the `Ack` module. But indeed it looks like it's doing similar ! 

I have added a commit that uses the `pushack` function as suggested. 
Now the `Segment.Rx` module is functorized against the `Ack.M` signature.